src/foam/foamlink/FoamlinkExec.js

```diff
@@ -0,0 +1,217 @@
+foam.CLASS({
+  package: 'foam.foamlink',
+  name: 'FoamlinkExec',
+  documentation: `
+    FoamlinkExec checks for foamlink.js in the current directory, then
+    begins following the instructions to produce an instance of
+    FoamlinkData.
+  `,
+  properties: [
+    ['repositories_', {}],
+    ['ignorePaths_', {}],
+    ['injectFiles_', {}],
+    ['reservedNames_', [
+      'files.js', 'files2.js', 'classes.js'
+    ]],
+    {
+      name: 'results',
+      class: 'FObjectProperty',
+      factory: function() {
+        return foam.foamlink.FoamlinkData.create();
+      }
+    }
+  ],
+  methods: [
+    function init() {
+      this.nodejs_ = {};
+      this.require_('path');
+      this.require_('fs');
+    },
+    function require_(name, asName) {
+      Object.defineProperty(this.nodejs_, asName || name, {
+        value: require(name),
+        writable: false
+      });
+    },
+    function exec(path) {
+      var self = this;
+      path = path || process.cwd();
+
+      return new Promise((resolve, reject) => {
+        this.processFoamlinkFile(path, true);
+
+        var walkPromises = [];
+        for ( repoPath in this.repositories_ ) {
+          walkPromises.push(this.startWalking(repoPath, reject));
+        }
+
+        Promise.all(walkPromises).then(resolve);
```

Comment:
Do we need a `return` here?

Reply:
This will be okay because calling `resolve` is equivalent to returning a promise